astropy/table/column.py

```diff
@@ -1803,6 +1803,12 @@ def insert(self, obj, values, mask=None, axis=0):
 
         return out
 
+    def convert_unit_to(self, new_unit, equivalencies=[]):
+        # This is a workaround to fix gh-9521
+        super().convert_unit_to(new_unit, equivalencies)
+        self._basedict["_unit"] = new_unit
```

Comment:
Yes, `MaskedArray` is really annoying! I think what you have here is about as efficient as it can be.